sympy/physics/units/quantities.py

```diff
@@ -88,6 +88,10 @@ def _eval_is_positive(self):
     def _eval_is_constant(self):
         return self.scale_factor.is_constant()
 
+    def _eval_Abs(self):
+        return self.__class__(self.name, self.dimension,
```

Comment:
I have discovered that while trying to solve the problem with missing assumptions on `Quantity`. I can change it to `self.func`, but I would still strongly advocate for using `self.__class__` instead as it keeps the inheritance working correctly which I have started testing in #12919.